gcloud-java-dns/src/test/java/com/google/cloud/dns/DnsBatchTest.java

```diff
@@ -0,0 +1,632 @@
+  private static final Dns.ProjectOption PROJECT_FIELDS =
+      Dns.ProjectOption.fields(Dns.ProjectField.QUOTA);
+  private static final Dns.ZoneListOption[] ZONE_LIST_OPTIONS = {
+      Dns.ZoneListOption.pageSize(MAX_SIZE), Dns.ZoneListOption.pageToken(PAGE_TOKEN),
+      Dns.ZoneListOption.fields(Dns.ZoneField.DESCRIPTION),
+      Dns.ZoneListOption.dnsName(DNS_NAME)};
+  private static final ProjectInfo PROJECT_INFO = ProjectInfo.builder().build();
+  private static final Dns.RecordSetListOption[] RECORD_SET_LIST_OPTIONS = {
+      Dns.RecordSetListOption.pageSize(MAX_SIZE),
+      Dns.RecordSetListOption.pageToken(PAGE_TOKEN),
+      Dns.RecordSetListOption.fields(Dns.RecordSetField.TTL),
+      Dns.RecordSetListOption.dnsName(DNS_NAME),
+      Dns.RecordSetListOption.type(RecordSet.Type.AAAA)};
+  private static final RecordSet RECORD_SET =
+      RecordSet.builder("Something", RecordSet.Type.AAAA).build();
+  private static final ChangeRequestInfo CHANGE_REQUEST_PARTIAL = ChangeRequestInfo.builder()
+      .add(RECORD_SET)
+      .build();
+  private static final String CHANGE_ID = "some change id";
+  private static final ChangeRequestInfo CHANGE_REQUEST_COMPLETE = ChangeRequestInfo.builder()
+      .add(RECORD_SET)
+      .startTimeMillis(123L)
+      .status(ChangeRequest.Status.PENDING)
+      .generatedId(CHANGE_ID)
+      .build();
+  private static final Dns.ChangeRequestListOption[] CHANGE_LIST_OPTIONS = {
+      Dns.ChangeRequestListOption.pageSize(MAX_SIZE),
+      Dns.ChangeRequestListOption.pageToken(PAGE_TOKEN),
+      Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.STATUS),
+      Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING)};
+  private static final Dns.ChangeRequestOption CHANGE_GET_FIELDS =
+      Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS);
+  private static final List<ResourceRecordSet> RECORD_SET_LIST = ImmutableList.of(
+      RECORD_SET.toPb(), RECORD_SET.toPb(), RECORD_SET.toPb(), RECORD_SET.toPb());
+  private static final List<Change> CHANGE_LIST = ImmutableList.of(CHANGE_REQUEST_COMPLETE.toPb(),
+      CHANGE_REQUEST_COMPLETE.toPb(), CHANGE_REQUEST_COMPLETE.toPb());
+  private static final List<ManagedZone> ZONE_LIST = ImmutableList.of(ZONE_INFO.toPb(),
+      ZONE_INFO.toPb());
+  private static final GoogleJsonError GOOGLE_JSON_ERROR = new GoogleJsonError();
+
+  private DnsOptions optionsMock;
+  private DnsRpc dnsRpcMock;
+  private RpcBatch batchMock;
+  private DnsBatch dnsBatch;
+  private Dns dns = EasyMock.createStrictMock(Dns.class);
+
+  @Before
+  public void setUp() {
+    optionsMock = EasyMock.createMock(DnsOptions.class);
+    dnsRpcMock = EasyMock.createMock(DnsRpc.class);
+    batchMock = EasyMock.createMock(RpcBatch.class);
+    EasyMock.expect(optionsMock.rpc()).andReturn(dnsRpcMock);
+    EasyMock.expect(dnsRpcMock.createBatch()).andReturn(batchMock);
+    EasyMock.replay(optionsMock);
```

Comment:
Nit: could combine this and replay calls below into one replay call using varargs, right?